### DISCUSSION TITLE
The different Firmwares, and what changed with them.

### DISCUSSION THREAD

**Discussion by VE7NGK** [df8oe/UHSDR#1890] [Q&A] [answered]
Hi Everyone,
    I've had my RS-918 for about a year and a half, and enjoy it very much. It came with Bootloader 4.1.2 and Firmware D2.9.42. Some of the stuff that I've been able to figure out about my radio is that it a flash size of 1024kb, and RAM of 192kb. Now, I'm not wanting to take away from everyone's enjoyment of all the digital modes, and everything that this beautiful radio can do, and I sure don't want to come across as a broken record when it comes to asking questions regarding FreeDV. But, with that said, the later is why I bought my radio, and I am hoping to put it to good use when having QSOs with my brother, and hoping some other operators will join in. (Currently thinking about buying him a radio for his coming birthday.)
   But here's the question, from the research that I've been able to do, it seems that the FreeDV used by this radio is the 1600 mode. With the soon to come 2020 version, and the 700d version out there in the wild, I was wondering if there is a firmware that has the 700D mode and if there are any plans to bring the 2020 version to this radio.
   I am aware that some older radios needed to drop FreeDV all together to make other digital modes work, but is there a project in place that might drop other digital modes and function and preserve FreeDV on the older radios with the newer (ish) FreeDV modes?
   Thanks for your time and consideration,
73's -Tom

**Comment by db4ple** [comment]
Hi  Tom,

this has been discussed a couple of times in various places: the new modes require more computational power than available in the mcHF and its clones. That will not change. 
UHSDR itself does support 700D but you will need a F7 or H7 based TRX (OVI40, Sparrow, ... ) to run 700D.
LPCnet based modes like 2020 are again too resource-intensive for any UHSDR  based TRX, And most likely will remain so unless UHSDR gets ported to even more powerful processors.

**Comment by VE7NGK** [comment]
Thank you so much! I had seen several posts regarding this on different places on the web but your explanation finally answers my question. But it leaves me with one more... I know that there is the FreeDV text message option, and I remember seeing somewhere that there was a firmware that let you enter a message, I guess your call sign, or location, or something along those lines. Was this for the mcHF based units and if so, is that in the most current version, or is there a specific one that I would be looking for? Thanks again! :)

**Comment by df8oe** [comment, marked as answer]
At the moment there is no possibility to personalize the text message.

**Reply by VE7NGK** [reply]
OK, thanks.

**Comment by Lotpe4** [comment]
I have the same radio and I have such a problem that I cannot update it, I try to play the software and it is a fiasco.[](url)
![IMG_20210314_194255 1](https://user-images.githubusercontent.com/62907767/111355284-84b39280-8687-11eb-8ed8-40d69a1d0e2f.jpg)
![IMG_20210311_223734 1](https://user-images.githubusercontent.com/62907767/111355366-985ef900-8687-11eb-9718-83e54473c462.jpg)

**Reply by db4ple** [reply]
@Lotpe4 : Please explain what your problem this, so that we can understand it. And please consider opening a separate discussion if your problem is not related to the original problem of this discussion.

Thanks!

**Comment by df8oe** [comment]
No comment.
![2021-03-16_18-52](https://user-images.githubusercontent.com/4942727/111357152-53d45d00-8689-11eb-9e76-73bf08dc5d81.png)

**Reply by Lotpe4** [reply]
I just have a problem with the MCH 918 radio update, I can't update it. I put the pendrive into the USB and it does not go Latest UHSDR Firmware version is 2.12.0

**Reply by Lotpe4** [reply]
I pressed the button, called the first line and it stands and the green diode on the radio is blinking Then I put the pendrive into the usb and it goes to the end and nothing updates

**Reply by db4ple** [reply]
Make your life easier by following the instructions EXACTLY AS DOCUMENTED:
  
https://github.com/df8oe/UHSDR/blob/active-devel/mchf-eclipse/useful_manuals/bootloader_doc.md#flashing-new-firmware-and-saving-old-firmware

**Comment by Lotpe4** [comment]
I managed to have a problem with the pendrive because he did not see him until I put the adapter and SD card in and the problem was over.

**Reply by df8oe** [reply]
Your picture I have commented above tells other story...
I can read there "USB Drive dectected." what definitely means *your drive was seen and recognized'
Next Saving of old firmware was finished successfully. That definitely means your stick is writeable for UHSDR.
Then the magic line "Skipping firmware update." appears which only can have one (!!) cause: *you released the BAND- button* before the message "Firmware will be updated" appeared*. It was an user - not a hardware error...
